Intel fix

crees opened the pull request:
Complete the Intel graphics fixes for those that initgfx decides need drm-kmod-legacy.

Nothing needs this, it's deprecated and will be removed when 11.4 is no longer supported.

The 'new' drm-kmod supports all of them properly.

crees committed:
Lock the special pkgs so they aren't overwritten

probonopd commented:
Thank you very much @crees.

Since we would like to not diverge `initgfx` from ustream, woud you mind sending a PR with these `initgfx` changes to https://github.com/nomadbsd/NomadBSD/ which is upstream for `initgfx`?

Once it is merged there, I'd like to use their new `initgfx` here.

---

https://nomadbsd.org/index.html#1.4 states

> Bad news for i386 users: Since drm-legacy-kmod is now obsolete, there is no i386 support for accelerated graphics for Intel and ATI/AMD cards anymore.

So this means that FreeBSD 12.x is the last one that is properly usable one on 32-bit Intel machines? I do care about 32-bit machines, especially those with Intel GPUs (e.g., first-gen Atom netbooks).

crees committed:
Revert "Fix for FreeBSD 12"

This was fixed differently upstream without relying on version numbers

This reverts commit 6dae2901c7df20fd76986d9a7cfa5948867a460b.

crees committed:
Revert "Try to fix #38"

Upstream fixed this differently

This reverts commit 11b6c0b5a50b37807ebddce62c5b36a88d6bd759.

crees committed:
Sync with upstream

This specifically fixes #38, and also updates the logic on Intel devices

commit f5931e3cf10aa2bef72a940e05845dd62d5a84e8
Author: Marcel <mk@freeshell.de>
Date:   Fri Mar 5 17:10:23 2021 +0100

    Add support for VMware.

commit 1fa9fe562bd9ca20765ef8d048d089d31d6d0fc1
Author: Marcel <mk@freeshell.de>
Date:   Fri Mar 5 17:09:57 2021 +0100

    Remove redundant line.

commit c4a44d086cb61b6a71958ca071eae56c56a778eb
Author: Marcel <mk@freeshell.de>
Date:   Sat Feb 20 10:35:03 2021 +0100

    Mounting / with -o sync doesn't seem to significantly reduce filesystem damage after a crash.

commit 7f681067f445526c9a0e69be4d5dca3b3967a876
Author: Marcel Kaiser <mk@freeshell.de>
Date:   Tue Feb 16 22:46:09 2021 +0100

    Change probe order. Since most Intel CPUs today have an IGPU, it is likely that initgfx will detect and use the IGPU before probing for an ATI/AMD graphics card.

commit c663c0073ed0b33c48863eef85f3c540566cc2a1
Author: Marcel <mk@freeshell.de>
Date:   Mon Feb 8 18:57:03 2021 +0100

    Temporarily mount / with -o sync which can reduce filesystem damage on crash.

commit 78a6567d72ba267f3eb58be5eb03135124ea77c8
Author: Marcel <mk@freeshell.de>
Date:   Sun Feb 7 20:53:02 2021 +0100

    Add an option to the boot menu to disable automatic graphics detection.

commit 750f46215cb97d25247414e04a814a757eb6a825
Author: Marcel <mk@freeshell.de>
Date:   Sun Jan 3 20:27:39 2021 +0100

    Change get_device_info() to also work under FreeBSD>=13, where pciconf outputs vendor and device separately

commit 1277288a1b779549e755d0b75a08e873c97a7049
Author: Marcel <mk@freeshell.de>
Date:   Mon Dec 28 22:14:05 2020 +0100

    Instead of writing to rc.conf directly, use a temporary file. Copy the temporary file back to /etc/rc.conf when done. This should prevent file corruption in case of a crash (Hi, Nvidia!)

commit e5bb8c1abf0433bd41d7dc4f8a6455a64084dfcf
Author: Marcel <mk@freeshell.de>
Date:   Sat Dec 26 23:19:28 2020 +0100

    Decrease delay

commit 7a3d2791ca1d3a169373e761d8f5ff1d69f65e05
Author: Marcel <mk@freeshell.de>
Date:   Sat Dec 26 21:26:10 2020 +0100

    - When testing X use xmessage to show a message for 10s. If xmessage is not available, sleep for 10s.
    - Copy the Nvidia files to /nvidia/.../modules/{drivers,extensions}
    - Omit copying vdpau directory. It it under the dynamic link path and will be found.
    - Make sure the symlinks to /nvidia/... under /usr/local/.../modules/{drivers,extensions} are still present. Recreate them if they were overwritten.
    - Waiting after loading the Nvidia module doesn't seem to have an impact on the crashing problem with the 340 driver. Remove it.
    - Change delay after testing X back to 3s.

commit 986fc3d63d239ec17befac495ce6c18b71183941
Author: Marcel <mk@freeshell.de>
Date:   Sat Dec 26 10:40:05 2020 +0100

    - Add a __wait() function.
    - Wait 10s after loading the nvidia module, and wait 10s after the Xorg test to prevent spontaneous reboots caused by the nvidia (340) module.

commit d5c836d60c72793bfef24c95af83926e294cc630
Author: Marcel <mk@freeshell.de>
Date:   Mon Dec 14 17:27:51 2020 +0100

    Remove drm-legacy-kmod, which is obsolete, and expires on 2020-12-31.

crees commented:
I'll test this once the build is finished and let you know if it works.  Do you expect i386 builds to just work?  I can try it out on my 2006 Macbook (would be wonderful to have this serviceable again!)

probonopd commented:
> Do you expect i386 builds to just work? I can try it out on my 2006 Macbook (would be wonderful to have this serviceable again!)

It's been a while since I last made an i386 build. I expect it to work for desktops **other** than helloSystem, because for helloSystme we currently have builds of some of our custom components (Menu, Filer, Dock,...) only for amd64 at the moment.

I think it is OK for now if it stays this way (=i386 should continue to work for non-helloSystem builds), and we can tackle non-amd64 builds of helloSystem later on).

probonopd commented:
Thanks @crees. Please let me know once you think this is ready for merging.
